### DISCUSSION TITLE
Custom Autocomplete While In SSH

### DISCUSSION THREAD

**Discussion by AwesomeIronman** [kovidgoyal/kitty#3326] [Ideas]
We can't use our aliases while SSHing servers
So can we have a custom shortcut / keybinding (ctrl + shift + tab) which will autocomplete
Certain words based on a kitty-dictionary.conf like new file

For eg. we are in SSH
we type $ run_build
and hit (ctrl + shift + tab) then 'run_build' will get replaced with $ prepare for build && run build
or something like that

**Comment by kovidgoyal** [comment]
Sure you can use your aliases, simply copy the alias definitions to the server.

**Reply by AwesomeIronman** [reply]
I'm not allowed to do that.

**Comment by kovidgoyal** [comment]
Well you can define aliases using the kitty send_text command to send arbitrary text on keypresses.

**Reply by AwesomeIronman** [reply]
So far i'm only aware of bash/zsh Aliases which won't work in ZSH and I'm not allowed to write to those files

How do we create Aliases using send_text command?
Send_text is supposed to just send any command to any tab/window, right?

**Reply by AwesomeIronman** [reply]
Understood.
`map alt+s send_text all \x13 `
I don't find this way much convenient though

**Reply by kovidgoyal** [reply]
Well that way is all a terminal can do, since it doesnt have any way to control processes running ona  remote machine. I dont know what "not allowed" means for you, but look into https://github.com/romkatv/zsh4humans  which automatically makes your zsh config available on the remote.

**Reply by AwesomeIronman** [reply]
Got it, thanks.
